diff --git a/config/application.rb b/config/application.rb
index 9676441d..8179d879 100644
--- a/config/application.rb
+++ b/config/application.rb
@@ -65,9 +65,10 @@ module EarlyYearsFoundationRecovery
     config.contentful_environment             = ENV.fetch('CONTENTFUL_ENVIRONMENT', credentials.dig(:contentful, :environment))
 
     # Gov One
-    config.gov_one_base_uri    = credentials.dig(:gov_one, :base_uri)
-    config.gov_one_private_key = credentials.dig(:gov_one, :private_key)
-    config.gov_one_client_id   = credentials.dig(:gov_one, :client_id)
+    config.gov_one_base_uri = ENV.fetch('GOV_ONE_BASE_URI', credentials.dig(:gov_one, :base_uri))
+    config.gov_one_private_key = ENV['GOV_ONE_PRIVATE_KEY_PATH'].present? ?
+      File.read(ENV.fetch('GOV_ONE_PRIVATE_KEY_PATH')) : credentials.dig(:gov_one, :private_key)
+    config.gov_one_client_id = ENV.fetch('GOV_ONE_CLIENT_ID', credentials.dig(:gov_one, :client_id))
 
     # Sentry
     config.sentry_dsn = ENV.fetch('SENTRY_DSN', '#SENTRY_DSN_env_var_missing')
diff --git a/app/services/gov_one_auth_service.rb b/app/services/gov_one_auth_service.rb
index 4fa2f011..8bf024c9 100644
--- a/app/services/gov_one_auth_service.rb
+++ b/app/services/gov_one_auth_service.rb
@@ -92,7 +92,7 @@ class GovOneAuthService
   def build_http(address)
     uri = URI.parse(address)
     http = http_client.new(uri.host, uri.port)
-    http.use_ssl = true
+    http.use_ssl = uri.scheme == 'https'
     [uri, http]
   rescue StandardError => e
     Rails.logger.error "GovOneAuthService.build_http: #{e.message}"
